src/dsmr/parser.h

```diff
@@ -215,8 +215,14 @@ struct NumParser {
     // If a unit was passed, check that the unit in the messages
     // messages the unit passed.
     if (unit && *unit) {
-      if (num_end >= end || *num_end != '*')
-        return res.fail(F("Missing unit"), num_end);
+      
+      if (num_end >= end || *num_end != '*') {
+        if (value != 0)
+          return res.fail(F("Missing unit"), num_end);
+        else
+          return res.succeed(value).until(num_end + 1); // Skip )
```

Comment:
This should not return `res.succeed` here, since it should just skip the unit and then continue parsing (i.e. still parse/check for the trailing `)`).

So this might also need a bit refactoring so if there is no unit but also no `)` (so `num_end >= end=) then an error should still be returned, even if `value = 0` (since then the line is truncated, not just the unit is missing).